libsolidity/interface/CompilerStack.h

```diff
@@ -216,11 +217,11 @@ class CompilerStack: boost::noncopyable
 
 	/// Parses and analyzes all source units that were added
 	/// @returns false on error.
-	bool parseAndAnalyze();
+	bool parseAndAnalyze(State _stopAfter = State::CompilationSuccessful);
```

Comment:
Referring to my above comment, I think this function should not have that parameter, and callers that needs to stop after parse should then use `compile(stopAfterStage)` or just call parse() directly?